monix-catnap/shared/src/main/scala/monix/catnap/Semaphore.scala

```diff
@@ -146,53 +129,7 @@ final class Semaphore[F[_]] private (provisioned: Long, ps: PaddingStrategy)(
     *
     * @param n number of permits to release - must be >= 0
     */
-  def releaseN(n: Long): F[Unit] =
-    underlying.releaseN(n)
-
-  /** Releases a permit, returning it to the pool.
-    *
-    * If there are consumers waiting on permits being available,
-    * then the first in the queue will be selected and given
-    * a permit immediately.
-    *
-    * @see [[withPermit]], the preferred way to acquire and release
-    */
-  override def release: F[Unit] = releaseN(1)
-
-  /** Returns a new task, ensuring that the given source
-    * acquires an available permit from the semaphore before
-    * it is executed.
-    *
-    * The returned task also takes care of resource handling,
-    * releasing its permit after being complete.
-    *
-    * @param fa is an effect to execute once the permit has been
-    *        acquired; regardless of its result, the permit is
-    *        released to the pool afterwards
-    */
-  def withPermit[A](fa: F[A]): F[A] =
-    withPermitN(1)(fa)
```

Comment:
`withPermit` is gone from CE3 semaphore, _and_ RC2 added a `mapK` requirement. If we want to extend CE3 semaphore, we need to provide transformation for any `MonadCancel[F, _]`, but that's not enough to do `permit.surround` (Resource.use needs `MonadCancel[F, Throwable]` specifically). We can still have the syntax  with a slightly different signature than before (requiring implicit `MonadCancel[F, Throwable]`), I'm just not sure if we want to keep it.